json-schema-form.d.ts

```diff
@@ -0,0 +1,42 @@
+/**
+ * Shorthand to lookup for keys with `x-jsf-*` preffix.
+ */
+export function pickXKey(node: Object, key: 'presentation' | 'error-message'): Object | undefined;
+
+type JSConfig = {
+  initialValues?: Record<string, unknown>;
+  strictInputType?: boolean;
+  customProperties?: {
+    description: Function | string;
+    [key: string]: unknown;
+  };
+  inputTypes?: {
+    errorMessage: Record<string, unknown>;
+  };
+};
+
+type Fields = Record<string, unknown>[]; // TODO: We don't know what type we have here, we need to investigate
+
+type $TsFixMe = any;
+
+/**
+ * Returns the Yup schema structure of given fields.
+ * These fields must be the same from
+ * const { fields } = createHeadlessForm()
+ */
+export function buildCompleteYupSchema(fields: Fields, config: JSConfig): $TsFixMe; // TODO: We don't what yup returns here, we'll fix it later
```

Comment:
extra note:

```suggestion
export function buildCompleteYupSchema(fields: Fields, config: JSConfig): $TsFixMe; // TODO: We need to update Yup to 1.0 which supports TS.
```

Reply:
Thanks, I didn't know what version started to support typescript

Reply:
Ah, I was wrong, afterall it's after [v0.32](https://github.com/jquense/yup/blob/master/CHANGELOG.md#0320-2020-12-03)